shared/ui/core/src/main/java/com/ivy/ui/FormatMoneyUseCase.kt

```diff
@@ -3,27 +3,43 @@ package com.ivy.ui
 import android.content.Context
 import com.ivy.domain.features.Features
 import com.ivy.ui.time.DevicePreferences
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
 import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
 import java.text.DecimalFormat
 import java.text.DecimalFormatSymbols
 import javax.inject.Inject
+import javax.inject.Singleton
 
-class FormatMoneyUseCase @Inject constructor(
+@Singleton
+class FormatMoneyUseCaseImpl @Inject constructor(
     private val features: Features,
     private val devicePreferences: DevicePreferences,
     @ApplicationContext private val context: Context
-) {
+) : FormatMoneyUseCase {
+
+    @EntryPoint
+    @InstallIn(SingletonComponent::class)
+    interface FormatMoneyUseCaseEntryPoint {
+        @Singleton
+        fun FormatMoneyUseCase(): FormatMoneyUseCase
+    }
 
     private val locale = devicePreferences.locale()
     private val withoutDecimalFormatter = DecimalFormat("###,###", DecimalFormatSymbols(locale))
-    private val withDecimalFormatter = DecimalFormat("###,###.00", DecimalFormatSymbols(locale))
+    private val withDecimalFormatter = DecimalFormat("###,##0.00", DecimalFormatSymbols(locale))
 
-    suspend fun format(value: Double): String {
+    override suspend fun format(value: Double): String {
         val showDecimalPoint = features.showDecimalNumber.isEnabled(context)
 
         return when (showDecimalPoint) {
             true -> withDecimalFormatter.format(value)
             false -> withoutDecimalFormatter.format(value)
         }
     }
-}
\ No newline at end of file
+}
+
+interface FormatMoneyUseCase {
```

Comment:
What's the purpose of this interface?

Reply:
I wanted to use `@Binds` instead of `@Provides` for corresponding dependency. Using @Provides now and so removed it.